src/semaphores_map.rs

```diff
@@ -0,0 +1,26 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::sync::Mutex;
+use tokio::sync::Semaphore;
+
+pub struct SemaphoresMap {
+    permits: usize,
+    inner: Mutex<HashMap<String, Arc<Semaphore>>>,
```

Comment:
We must use async mutex in async context!

Reply:
I thought i did use tokio mutex, fiixing it